src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -762,8 +877,51 @@ internal override void Complete()
                     }
 
                     this.Command.Context.SetVariable(SpecialVariables.LastExitCodeVarPath, _nativeProcess.ExitCode);
-                    if (_nativeProcess.ExitCode != 0)
-                        this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+                    if (_nativeProcess.ExitCode == 0)
+                    {
+                        return;
+                    }
+
+                    this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+
+                    bool nativeCommandUseErrorActionPreference = ExperimentalFeature.IsEnabled(ExperimentalFeature.PSNativeCommandErrorActionPreferenceFeatureName)
+                        && this.Command.Context.GetBooleanPreference(
+                            SpecialVariables.PSNativeCommandUseErrorActionPreferenceVarPath,
+                            InitialSessionState.DefaultPSNativeCommandUseErrorActionPreference,
+                            out _);
+
+                    if (!nativeCommandUseErrorActionPreference)
+                    {
+                        return;
+                    }
+
+                    ActionPreference errorActionPref =
+                        this.Command.Context.GetEnumPreference(
+                            SpecialVariables.ErrorActionPreferenceVarPath,
+                            ActionPreference.SilentlyContinue,
+                            out _);
+
+                    if (errorActionPref != ActionPreference.Ignore)
+                    {
+                        const string errorId = "ProgramFailedToComplete";
```

Comment:
Code convention for consts.
```suggestion
                        const string ErrorId = "ProgramFailedToComplete";
```

Reply:
Didn't we discuss that for local consts camelCase is preferred?